cinn/hlir/framework/graph_compiler.h

```diff
@@ -94,6 +95,16 @@ class GraphCompiler final {
     std::string attached_code                    = "";
     bool with_instantiate_variables              = false;
     bool with_buffer_handle_instruction_inserted = false;
+    bool remove_unused_variables                 = true;
+    // nodes group, it may come from the result of op fusion or graph tuning.
+    // nodes in a group will be built into an Instruction
+    std::vector<std::vector<Node*>> groups;
```

Comment:
It seems you store some results of tuning in `CompileOptions`, it may not be good because the struct is named as an option. Is there other way to pass/store these results?

Reply:
I considered that and why I chosed this way was to be compatible with current `Build` interface. In the future refactor , I will update the interface with more meaningful parameter structs and names.

Reply:
Ok, I will approve this PR and we shall mark it as TODO.